### PR TITLE
Paybill Validation and Settlement in mpesa-connector

### PR DESCRIPTION
**Pull request by truthfool**
- Receives paybill request payload from mpesa
- sending only the validation payload and the required fields channel connector.
- Sending the confirmation/settlement payload to paygops
- Resuming the workflow after verifying mpesaTransactionId and workflow transaction id
@avikganguly01 @ankita10r @fynmanoj

Please do not merge it yet.

### CODE REVIEW

File: src/main/java/org/mifos/connector/mpesa/dto/ChannelRequestDTO.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override
+    public String toString() {
+        return "{" +
+                "primaryIdentifier:" + primaryIdentifier +
+                ", secondaryIdentifier:" + secondaryIdentifier +
+                ", customData:" + customData +
+                "}";
+    }
+
+    public JSONObject getPrimaryIdentifier() {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mpesa/dto/ChannelRequestDTO.java:25]: Use Lombok  here.  no need to add boiler plate code

File: src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+//    {
+//        "TransactionType":"Pay Bill",
+//        "TransID":"RKTQDM7W6S",
+//        "TransTime":"20191122063845",
+//        "TransAmount":"10",
+//        "BusinessShortCode":"600638",
+//        "BillRefNumber":"A123",
+//        "InvoiceNumber":"",
+//        "OrgAccountBalance":"49197.00",
+//        "ThirdPartyTransID":"",
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java:15]: remove comments

File: src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    @JsonProperty("MSISDN")
+    private String msisdn;
+
+    @JsonProperty("FirstName")
+    private String firstname;
+
+    public PaybillRequestDTO() {
+    }
+
+    public PaybillRequestDTO(String transactionType, String transactionID, String transactionTime, String transactionAmount, String shortCode, String billRefNo, String invoiceNumber, String accountBalance, String thirdPatrytransactionID, String msisdn, String firstname) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java:55]: use lombok

File: src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java [hunk 4]
```diff
@@ -20,29 +25,85 @@ public class MpesaUtils {
...
+        }
+        return null;
+    }
+
+    public static ChannelRequestDTO convertPaybillPayloadToChannelPayload(PaybillRequestDTO paybillRequestDTO, String amsName) {
+        String foundationalId = "";
+        String accountID = "";
+        // Mapping primary and secondary Identifier
+        JSONObject primaryIdentifier = new JSONObject();
+        if (amsName.equalsIgnoreCase("paygops")) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java:42]: "paygops" this should be an ENUM

File: src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java [hunk 5]
```diff
@@ -61,25 +122,25 @@ public static String maskString(String strText) {
...
-        for(int i = 0; i < maskLength; i++){
+        for (int i = 0; i < maskLength; i++) {
             sbMaskString.append(maskChar);
         }
 
         return strText.substring(0, start) + sbMaskString + strText.substring(start + maskLength);
     }
 
-    public static void main(String[] args)  {
+    public static void main(String[] args) {
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java:143]: Why do we need a main method here?  Donot sneak in main methods

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java (hunk 2 above)
Comment (truthfool) [reply src/main/java/org/mifos/connector/mpesa/dto/PaybillRequestDTO.java:15]: Its a comment for request payload for reference.

File: src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java (hunk 5 above)
Comment (truthfool) [reply src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java:143]: I have not added that. It was there previously.

Comment (fynmanoj) [reply src/main/java/org/mifos/connector/mpesa/utility/MpesaUtils.java:143]: Lets remove that

**Pull request merged**